Classes/common/HTTP/CDTHTTPInterceptorContext.m

```diff
@@ -24,11 +30,18 @@ -(instancetype)init {
 }
 
 - (instancetype)initWithRequest:(NSMutableURLRequest *)request {
+    return [self initWithRequest:request response:nil];
+}
+
+- (nullable instancetype)initWithRequest:(nonnull NSMutableURLRequest *)request
+                                response:(nullable NSURLResponse *)response
+{
     NSParameterAssert(request);
     self = [super init];
     
     if (self) {
         _request = request;
+        _shouldRetry = NO;
```

Comment:
You never assign the response; this method isn't used and should probably be removed.